torch/csrc/cuda/nccl.cpp

```diff
@@ -329,7 +329,10 @@ bool is_available(TensorList tensors) {
 
 std::uint64_t version() {
 #if defined(NCCL_MAJOR)
-  return NCCL_MAJOR * 1000 + NCCL_MINOR * 100 + NCCL_PATCH;
+  constexpr std::uint64_t ver = (((uint64_t) NCCL_MAJOR) << 32) |
+                                (((uint64_t) NCCL_MINOR) << 16) |
+                                ((uint64_t) NCCL_PATCH);
+  return ver;
 #elif defined(USE_NCCL)
   return 1000;
```

Comment:
IIUC, this will be parsed as `0.0.1000` by `nccl.version`?

Reply:
good catch, this should be `1 << 32` in that case!